proof/invariant-abstract/AARCH64/ArchDetSchedSchedule_AI.thy

```diff
@@ -508,7 +510,7 @@ lemma dmo_scheduler_act_sane[wp]:
 lemma vgic_maintenance_irq_valid_sched[wp]:
   "\<lbrace>valid_sched and invs and scheduler_act_sane and ct_not_queued\<rbrace>
    vgic_maintenance
-   \<lbrace>\<lambda>rv. valid_sched\<rbrace>"
+   \<lbrace>\<lambda>rv. valid_sched :: det_state \<Rightarrow> _\<rbrace>"
```

Comment:
hang on, we can do dummy types in a type signature? how did you discover this?

Reply:
I think I already knew that it was possible, but this specific pattern was copied from rt.